src/wp-includes/formatting.php

```diff
@@ -5394,6 +5394,34 @@ function normalize_whitespace( $str ) {
  * @return string The processed string.
  */
 function wp_strip_all_tags( $text, $remove_breaks = false ) {
+	if ( is_null( $text ) ) {
+		return '';
+	}
+
+	if ( ! is_scalar( $text ) ) {
+		if ( WP_DEBUG ) {
```

Comment:
Oh darn... one more question... why is this wrapped in a `if ( WP_DEBUG )` ? I mean, previously, without this change, the warning would already show in PHP < 8.0 (and be a fatal in 8.0+), so why are we now hiding this warning when `WP_DEBUG` is turned off ?

Reply:
Pinging @peterwilsoncc, as I can't quite remember why this check is here. It's possible that I took it from elsewhere in Core, or that myself or Peter mentioned using it at some stage.

Reply:
I think it was taken from elsewhere in Core, there are three patterns I see used:

* wrap `trigger_error` per this code
* alter the level depending on the WP_debug setting: `WP_DEBUG ? E_USER_WARNING : E_USER_NOTICE`
* unwrapped

My inclination is to use one of the first two but don't have strong feelings about it.

Reply:
I'd prefer not to hide notices which should be fixed in plugins/themes (option 3). Too many devs don't appear to use `WP_DEBUG` and/or don't take note of notices/warnings etc, so hiding them behind a setting which is only used by experienced devs is not helping anyone.

Also note that in most cases, the WP default error configuration hides most of these anyway (though not necessarily warnings).

Either way, just my two pennies.

Reply:
What about the `WP_DEBUG ? E_USER_WARNING : E_USER_NOTICE` (option 2), which would still inform both?

Is there a benefit to **warning** devs to fix something, and **notifying** users that something is wrong? Or should they both be warned?

Reply:
As the guard fixes the issue while the script runs, isn't the goal to notify the user/dev that something was wrong but the script gracefully handled?

If that's the case I say just use `E_USER_WARNING` as we can't count on devs to have set `WP_DEBUG`.

Reply:
As the primary intention of this ticket changed to stop PHP 8.0+ throwing a fatal when receiving a non-scalar (non-`null`) value, I'm going to push a commit to remove the `WP_DEBUG` wrapper and just throw an `E_USER_WARNING` for consistency with pre-PHP 8.0 behaviour.

As `null` did not throw an `E_USER_WARNING`, nor a deprecation notice pre-PHP 8.0, I'm going to keep the special-case for `null` for now to maintain consistency with pre-PHP 8.0 behaviour/output (besides, Core has some work to do to stop calling `wp_strip_all_tags( null )` _itself_ 😱.

In doing so, my thinking is that for now, this function will behave as if PHP 8.0 was never released, until we can decide on a clear path forward for Core to resolve inconsistent handling of errors, type deprecations, and passing of undocumented types.